src/com/qamp/HarisJasarevic/homeworks/homework2/task3/Robot.java

```diff
@@ -0,0 +1,74 @@
+package com.qamp.HarisJasarevic.homeworks.homework2.task3;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Robot {
+
+    private int x;
+    private int y;
+    private int robotMoves;
+    private List<String> robotPositions;
+
+    public Robot() {
+        this.x = 0;
+        this.y = 0;
+    }
```

Comment:
This is a bug, you should always initialise all attributes, especially references to Objects, in this case `robotPositions`.

You should have done:
`this(0,0)`

This would work because it would use the other constructor.

The default value of `robotPositions` is `null`. 

So if I do:
```
Robot robot = new Robot();
robot.moveSouth();
```

This will throw a `NullPointerException`.

Reply:
Malo mi je pala koncentracija kod zadatka Robot, shavatio sam da sam loše krenuo i onda sam samo tražio način da se izvučem. Svakako, pregledao sam komentare pa ću ovo sve doraditi. Hvala Tarik.